src/java/org/apache/cassandra/index/IndexStatusManager.java

```diff
@@ -183,35 +212,81 @@ public synchronized void propagateLocalIndexStatus(String keyspace, String index
 
             // Don't try and propagate if the gossiper isn't enabled. This is primarily for tests where the
             // Gossiper has not been started. If we attempt to propagate when not started an exception is
             // logged and this causes a number of dtests to fail.
             if (Gossiper.instance.isEnabled())
             {
-                String newStatus = JsonUtils.JSON_OBJECT_MAPPER.writeValueAsString(states);
+                // Versions 5.0.0 through 5.0.2 use a much more bloated format that duplicates keyspace names
+                // and writes full status names instead of their numeric codes. If the minimum cluster version is
+                // unknown or one of those 3 versions, continue to propagate the old format.
+                CassandraVersion minVersion = Gossiper.instance.getMinVersion(1, TimeUnit.SECONDS);
+                String newSerializedStatusMap = shouldWriteLegacyStatusFormat(minVersion) ? JsonUtils.writeAsJsonString(statusMap) 
+                                                                                          : toSerializedFormat(statusMap);
+
                 statusPropagationExecutor.submit(() -> {
                     // schedule gossiper update asynchronously to avoid potential deadlock when another thread is holding
                     // gossiper taskLock.
-                    VersionedValue value = StorageService.instance.valueFactory.indexStatus(newStatus);
+                    VersionedValue value = StorageService.instance.valueFactory.indexStatus(newSerializedStatusMap);
                     Gossiper.instance.addLocalApplicationState(ApplicationState.INDEX_STATUS, value);
                 });
             }
         }
-        catch (Throwable e)
+        catch (Exception e)
         {
             logger.warn("Unable to propagate index status: {}", e.getMessage());
         }
     }
 
+    private static boolean shouldWriteLegacyStatusFormat(CassandraVersion minVersion)
+    {
+        return minVersion == null || (minVersion.major == 5 && minVersion.minor == 0 && minVersion.patch < 3);
+    }
+
+    /**
+     * Serializes as a JSON string the status of the indexes in the provided map.
+     * <p> 
+     * For example, the map...
+     * <pre>
+     * {
+     *     ks1.cf1_idx1=FULL_REBUILD_STARTED,
+     *     ks1.cf1_idx2=FULL_REBUILD_STARTED,
+     *     system.PaxosUncommittedIndex=BUILD_SUCCEEDED
+     * }
+     * </pre>
+     * ...will be converted to the string...
+     * <pre>
+     * {
+     *     "system": {"PaxosUncommittedIndex": 3},
+     *     "ks1": {"cf1_idx1": 1, "cf1_idx2": 1}
+     * }
+     * </pre>
+     */
+    public static String toSerializedFormat(Map<String, Index.Status> indexStatusMap)
+    {
+        Map<String, Map<String, Integer>> serialized = new HashMap<>();
+
+        for (Map.Entry<String, Index.Status> e : indexStatusMap.entrySet())
+        {
+            String keyspace = e.getKey().split("\\.")[0];
```

Comment:
```
String name = "trying.to.break.you";
schemaChange(String.format("CREATE TABLE %s.\"%s\"(pk int primary key)", KEYSPACE, name));
```

failed with

```
Caused by: org.apache.cassandra.exceptions.ConfigurationException: cql_test_keyspace.trying.to.break.you: Table name must not be empty, more than 48 characters long, or contain non-alphanumeric-underscore characters (got "trying.to.break.you")
```


so this feels fine... but you might be quoted (depending on where this is defined, trying to find)

`"Foo"."Bar"`

Reply:
ok think you don't have this issue as I was wrong on the format

`org.apache.cassandra.index.IndexStatusManager#getIndexStatus`

this takes the unquoted keyspace/index name

Reply:
```
String tbl = createTable("CREATE TABLE %s(pk int primary key, v int)");
        // I don't want to google the syntax
        var ddl = new CreateIndexDDL(CreateIndexDDL.Version.V2, CreateIndexDDL.SAI,
                                     Optional.of(new Symbol("Break.Things", Int32Type.instance)),
                                     new TableReference(Optional.of(KEYSPACE), tbl),
                                     Collections.singletonList(new Symbol("v", Int32Type.instance)),
                                     Collections.emptyMap());
        schemaChange(ddl.toCQL());
```

fails with

```
Caused by: org.apache.cassandra.exceptions.ConfigurationException: Illegal index name Break.Things
```